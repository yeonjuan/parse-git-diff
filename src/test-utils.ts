import Context from './context';

type Validator<Args extends any[] = any[]> = (...args: Args) => boolean;

function createBooleanTester<V extends Validator>(
  validator: V,
  result: boolean
) {
  return function validTester(cases: Parameters<V>[]): void {
    cases.forEach((c) => {
      expect(validator(...c)).toBe(result);
    });
  };
}

export function createValidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, true);
}

export function createInvalidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, false);
}

export function createContext(diff: string, initial: number = 1) {
  const context = new Context(diff);
  for (let i = 0; i < initial - 1; i++) {
    context.nextLine();
  }
  return context;
}
