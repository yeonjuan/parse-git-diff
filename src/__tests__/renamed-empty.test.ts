import { getFixture } from './test-utils';
import parseGitDiff from '../parse-git-diff';

describe('renamed-empty', () => {
  const fixture = getFixture('renamed-empty');

  it('parse `renamed-empty`', () => {
    expect(parseGitDiff(fixture)).toMatchSnapshot();
  });
});
