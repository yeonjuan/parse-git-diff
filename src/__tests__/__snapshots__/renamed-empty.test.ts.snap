// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renamed-empty parse \`renamed-empty\` 1`] = `
{
  "files": [
    {
      "chunks": [],
      "pathAfter": "src/renamed-empty",
      "pathBefore": "src/empty",
      "type": "RenamedFile",
    },
  ],
  "type": "GitDiff",
}
`;
